main.go

```diff
@@ -519,11 +565,21 @@ func productsInsertHelper(db *sql.DB, args []any, statement string, w *tabwriter
 	if err != nil {
 		return err
 	}
+	name, ok := args[0].(string)
```

Comment:
What if we just passed in a `p Product` object here instead of `args []any`? That way, we maintain type safety and it's clear to the caller what the argument types/order should be.